src/kubernetes.cc

```diff
@@ -1141,6 +1144,7 @@ void KubernetesReader::WatchNode(MetadataUpdater::UpdateCallback callback)
   } catch (const KubernetesReader::QueryException& e) {
     LOG(ERROR) << "No more node metadata will be collected";
   }
+  healthReporter.SetUnhealthy("node_callback");
```

Comment:
You're reporting health after the log in the case of the pod watch but before here. Let's be consistent and pick one.